charts/azuremonitor-containers/templates/omsagent-daemonset.yaml

```diff
@@ -179,6 +185,12 @@ spec:
          - mountPath: /etc/omsagent-secret
            name: omsagent-secret
            readOnly: true
+         {{- if and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.proxyCert) }}
+         - mountPath: /etc/ssl/certs/proxy-cert.crt
```

Comment:
ca cert can be changed (as part of cert rolling) post cluster creation in AKS, after which all our pods should be manually restarted to pick up the new cert . I see this in Ace's PR in AKS RP as well, which needs to be updated as well. https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp/pullrequest/5449520 -- Please include the team in PRs.

Reply:
> ca cert can be changed (as part of cert rolling) post cluster creation in AKS, after which all our pods should be manually restarted to pick up the new cert . I see this in Ace's PR in AKS RP as well, which needs to be updated as well. https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp/pullrequest/5449520 -- Please include the team in PRs.
> This will be taken care of the  AKS cert update workflow  for proxy CA cert (az aks update -> new node pool -> delete old node pool).

Reply:
Also looking thru ubuntu man pages, i see the recommended way to install custom ca certs seems using the tool 'update-ca-certs' which will also update the index list of certs in /etc/ssl/certs/ca-certificates.crt, and we drop the cert in  /usr/share/ca-certificates .

Reply:
> Also looking thru ubuntu man pages, i see the recommended way to install custom ca certs seems using the tool 'update-ca-certs' which will also update the index list of certs in /etc/ssl/certs/ca-certificates.crt, and we drop the cert in /usr/share/ca-certificates .

update-ca-certs after drop /usr/share/ca-certificates is other option instead of mounting the cert to /etc/ssl/certs/proxy-cert.crt.  since we are mounting the cert to /etc/ssl/certs/proxy-cert.crt so we dont need to run the update-ca-certs.

Reply:
why did we choose this option then ? Looks to me update-ca-certs also looks to ensure its a cert before adding to its index of certs...

Reply:
> why did we choose this option then ? Looks to me update-ca-certs also looks to ensure its a cert before adding to its index of certs...

>  As I I said, either options fine. I picked this option after talking folks who added in the Arc k8s, they opted this option to to work with distroless.

Reply:
i checked , and the only validation happening is base64 encoding. So you can wrap anything as base64 string and get that landed in /etc/ssl/proxycert.crt